<launch>
  <arg name="init_x" default="0.0" />
  <arg name="init_y" default="0.0" />
  <arg name="init_z" default="0.0" />
  <arg name="init_roll" default="0.0" />
  <arg name="init_pitch" default="0.0" />
  <arg name="init_yaw" default="0.0" />
  <arg name="to_trajectory_id" default="0" />
  <arg name="timestamp" default="0" />

  <!-- start trajectory with independent cartgrapher node -->
  <node name="cartographer_start_trajectory" pkg="cartographer_ros" type = "cartographer_start_trajectory" args="-configuration_directory $(find cartographer_ros_pr2)/configuration_files  -configuration_basename velodyne_spinal.lua -initial_pose '{to_trajectory_id = $(arg to_trajectory_id), timestamp = $(arg timestamp), relative_pose = { translation = { $(arg init_x), $(arg init_y), $(arg init_z) }, rotation = { $(arg init_roll), $(arg init_pitch), $(arg init_yaw) } } }'"/>
</launch>
